### PR TITLE
Hashan/add test

### PR DESCRIPTION
**Pull request by xxhashanxx**
Add test cases for stripe.hpp and change begin() and end() to const to support iterate over constant stripe objects as well as to support rbegin() and rend() which are const.

### CODE REVIEW

**Comment by xxhashanxx** [comment]
Unit testing completed. 
Fix few minor issues as well.

File: README.md [hunk 1]
```diff
@@ -441,3 +441,9 @@ future we may support other word sizes and input/output types.
 
 See `oned/simple8b.hpp` for full documentation. See `oned/examples/simple8b.cc`
 for more examples.
+
+### Build and Test Instructions
+
+Install Bazel and upgrade gcc ( 10 or above since you need c++20)
+from root directory run bazel build ...
```
Comment (gregulator) [inline README.md]: Use backticks for code snippets and use full sentences:
```suggestion
From root directory run `bazel build ...`.
```

File: README.md [hunk 2]
```diff
@@ -441,3 +441,9 @@ future we may support other word sizes and input/output types.
...
+for testing run bazel test ...
```
Comment (gregulator) [inline README.md]:
```suggestion
For testing, run `bazel test ...`.
```

File: oned/bench/chunks_bench.cc [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#include <benchmark/benchmark.h>
```
Comment (gregulator) [inline oned/bench/chunks_bench.cc:1]: Looks good! Thanks for adding this!

File: oned/bench/chunks_bench.cc [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+#include "oned/chunks.hpp"
+#include <vector>
+
+// Function to generate a test sequence
+std::vector<int> GenerateTestSequence(size_t size) {
+  std::vector<int> data(size);
+  for (size_t i = 0; i < size; ++i) {
+    data[i] = static_cast<int>(i); // Simple increasing sequence
+  }
```
Comment (gregulator) [inline oned/bench/chunks_bench.cc]: Optional: You can replace this for loop with:
```suggestion
  std::iota(data.begin(), data.end(), 0);  // Simple increasing sequence
```

You'll need a `#include <numeric>` at the top.

File: oned/chunks.hpp [hunk 5]
```diff
@@ -137,13 +137,22 @@ template <typename T> class Chunks {
     return reverse_iterator(begin());
   }
 
-  // Access the first chunk. Calling front on an empty Chunks results in
-  // undefined behavior.
-  constexpr Stripe<T> front() const { return (*this)[0]; }
+  class StripeProxy {
```
Comment (gregulator) [inline oned/chunks.hpp:140]: This is fine for now. However, this makes me think I'm doing something wrong with my Iterator if this is needed. I'll add it to my "todo" list to look into why we need this.

File: oned/chunks.hpp [hunk 6]
```diff
@@ -137,13 +137,22 @@ template <typename T> class Chunks {
...
+
+      auto begin() const { return (*chunks_)[index_].begin(); }
+      auto end() const { return (*chunks_)[index_].end(); }
+      operator Stripe<T>() const { return (*chunks_)[index_]; }
+  private:
+      const Chunks *chunks_;
+      size_t index_;
+  };
+
+  constexpr StripeProxy front() const { return StripeProxy(this, 0); }
```
Comment (gregulator) [inline oned/chunks.hpp:155]: All public methods should have doc comments. I think it is OK to just leave to old comments there:
```suggestion
  // Access the first chunk. Calling front on an empty Chunks results in
  // undefined behavior.
  constexpr StripeProxy front() const { return StripeProxy(this, 0); }
```

File: oned/chunks.hpp [hunk 7]
```diff
@@ -137,13 +137,22 @@ template <typename T> class Chunks {
...
+  constexpr StripeProxy back() const { return StripeProxy(this, size() - 1); }
```
Comment (gregulator) [inline oned/chunks.hpp:158]:
```suggestion
  // Access the last chunk. Calling back on an empty Chunks results in
  // undefined behavior.
  constexpr StripeProxy back() const { return StripeProxy(this, size() - 1); }
```

File: oned/stripe.hpp [hunk 8]
```diff
@@ -146,12 +146,13 @@ template <typename T> class Stripe {
 
   // Returns an iterator pointing to the first element of this span, or `end()`
   // if the span is empty.
-  constexpr iterator begin() noexcept { return Iterator(this, 0); }
+  //change this to const so we can iterate over const stripe objects
```
Comment (gregulator) [inline oned/stripe.hpp]: Comment not needed.
```suggestion
```

**Review by gregulator — DISMISSED** [review]
This is really fantastic! Thank you. Just a few small comments.

File: oned/bench/chunks_bench.cc (hunk 4 above)
Comment (xxhashanxx) [reply oned/bench/chunks_bench.cc]: Thanks a lot, First time learning this way

File: oned/chunks.hpp (hunk 5 above)
Comment (xxhashanxx) [reply oned/chunks.hpp:140]: Seems issue was with scope of (*this)[pos]. please let me know when you add fix for this.

**Comment by gregulator** [comment]
Looks good! Thank you.

**Review by gregulator — APPROVED** [review]

**Pull request merged**